libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m

```diff
@@ -289,8 +298,8 @@ - (BOOL) readAnalyticsPolicy {
 
 - (void) publishOnAppBackground {
 
-    // Publishing should only happen for the current user, not for all users signed in.
-    if (![self.userAccount.accountIdentity isEqual:[SFUserAccountManager sharedInstance].currentUser.accountIdentity]) {
+    // Publishing should only happen for the current user, and any anonymous stats, not for all users signed in.
+    if (self.userAccount != nil && ![self.userAccount.accountIdentity isEqual:[SFUserAccountManager sharedInstance].currentUser.accountIdentity]) {
```

Comment:
Seemed like this would be the place to publish unauthenticated analytics data (presuming publish actions were defined).  But I could also see making publishing of unauthenticated log lines the responsibility of the consumer too.  Thoughts?

Reply:
I would leave the publishing of unauthenticated events the responsibility of the consumer at this time, given that we have no generic strategy to publish at the moment.

Reply:
That seems reasonable to me.  I'll update/revert.